src/Tests/dotnet-workload-install.Tests/GivenNetSdkManagedWorkloadInstall.cs

```diff
@@ -0,0 +1,255 @@
+            var version = "6.0.100";
+            installer.InstallWorkloadPack(packInfo, new SdkFeatureBand(version));
+
+            var mockNugetInstaller = nugetInstaller as MockNuGetPackageInstaller;
+            mockNugetInstaller.InstallCallParams.Count.Should().Be(1);
+            mockNugetInstaller.InstallCallParams[0].ShouldBeEquivalentTo((new PackageId(packInfo.Id), new NuGetVersion(packInfo.Version)));
+            mockNugetInstaller.ExtractCallParams.Count.Should().Be(1);
+            mockNugetInstaller.ExtractCallParams[0].ShouldBeEquivalentTo((mockNugetInstaller.InstallCallResult[0], Path.Combine(dotnetRoot, "metadata", "temp", packInfo.Id + "-extracted")));
+
+            var installationRecordPath = Path.Combine(dotnetRoot, "metadata", "workloads", "InstalledPacks", packInfo.Id, packInfo.Version, version);
+            File.Exists(installationRecordPath).Should().BeTrue();
+
+            Directory.Exists(packInfo.Path).Should().BeTrue();
+        }
+
+        [Fact]
+        public void GivenManagedInstallItCanInstallSingleFilePacks()
+        {
+            var (dotnetRoot, installer, nugetInstaller) = GetTestInstaller();
+            var packInfo = new PackInfo("Xamarin.Android.Sdk", "8.4.7", WorkloadPackKind.Template, Path.Combine(dotnetRoot, "template-packs", "Xamarin.Android.Sdk.8.4.7.nupkg"));
+            var version = "6.0.100";
+            installer.InstallWorkloadPack(packInfo, new SdkFeatureBand(version));
+
+            (nugetInstaller as MockNuGetPackageInstaller).InstallCallParams.Count.Should().Be(1);
+            (nugetInstaller as MockNuGetPackageInstaller).InstallCallParams[0].ShouldBeEquivalentTo((new PackageId(packInfo.Id), new NuGetVersion(packInfo.Version)));
+            (nugetInstaller as MockNuGetPackageInstaller).ExtractCallParams.Count.Should().Be(0);
+
+            var installationRecordPath = Path.Combine(dotnetRoot, "metadata", "workloads", "InstalledPacks", packInfo.Id, packInfo.Version, version);
+            File.Exists(installationRecordPath).Should().BeTrue();
+
+            File.Exists(packInfo.Path).Should().BeTrue();
+        }
+
+        [Fact]
+        public void GivenManagedInstallItDetectInstalledPacks()
+        {
+            var (dotnetRoot, installer, _) = GetTestInstaller();
+            var packInfo = new PackInfo("Xamarin.Android.Sdk", "8.4.7", WorkloadPackKind.Sdk, Path.Combine(dotnetRoot, "packs", "Xamarin.Android.Sdk", "8.4.7"));
+            var version = "6.0.100";
+
+            // Mock installing the pack
+            Directory.CreateDirectory(packInfo.Path);
+
+            installer.InstallWorkloadPack(packInfo, new SdkFeatureBand(version));
+
+            _reporter.Lines.Should().Contain(string.Format(LocalizableStrings.WorkloadPackAlreadyInstalledMessage, packInfo.Id, packInfo.Version));
+        }
+
+        [Fact]
+        public void GivenManagedInstallItCanRollBackInstallFailures()
```

Comment:
Could we have coverage of what happens when multiple packs are supposed to be installed, and some install successfully but then one fails and they all need to be rolled back?

Reply:
That's what this test covers: https://github.com/dotnet/sdk/pull/16530/files#diff-c8bc89094b206b506f35f0cb9ac6e55591ae2468130efe0a19dab1810c0f0c1cR52, the first pack installs successfully, the second fails, and they are both rolled back.